pyproject_metadata/__init__.py

```diff
@@ -604,13 +603,15 @@ def _write_metadata(  # noqa: C901
         if self.auto_metadata_version != "2.1":
             for field in self.dynamic_metadata:
                 if field.lower() in {"name", "version", "dynamic"}:
-                    msg = f"Field cannot be set as dynamic metadata: {field}"
-                    raise ConfigurationError(msg)
+                    msg = f"Field cannot be set as dynamic metadata: {field!r}"
+                    errors.config_error(msg)
                 if field.lower() not in constants.KNOWN_METADATA_FIELDS:
-                    msg = f"Field is not known: {field}"
-                    raise ConfigurationError(msg)
+                    msg = f"Field is not known: {field!r}"
```

Comment:
I know I'm being pedantic, but I think it would be best to be consistent about the quotes used to delimit fields names. In most other places double quotes are used for field names.

Reply:
This is different; it's the value in `dynamic_metatata`, and it's not a field name, like `"project.description"`, it's a metadata name, like `'Summary'`. I think we should change the term "field" here.

Reply:
What about `"Metadata entry {field!r} cannot be declared dynamic"` and `"Unknown metadata entry {field!r} in dynamic declaration"` or something like this?

Reply:
I just wrote `Metadata name 'unknown' is not known, can't be used in dynamic metadata`, but I like `entry` better. And `declared dynamic`.